usr.sbin/devinfo/devinfo.c

```diff
@@ -83,21 +118,31 @@ print_resource(struct devinfo_res *res)
 	struct devinfo_rman	*rman;
 	bool			hexmode;
 	rman_res_t		end;
+	char			*safe_desc;
 
 	rman = devinfo_handle_to_rman(res->dr_rman);
 	hexmode =  (rman->dm_size > 1000) || (rman->dm_size == 0);
 	end = res->dr_start + res->dr_size - 1;
 
+	safe_desc = (char*) malloc(strlen(rman->dm_desc));
+	strcpy(safe_desc, rman->dm_desc);
```

Comment:
`strlcpy()` seems better here, with the previously calculated length of rman->dm_desc. maybe length + 1 (in malloc()) is needed (for the null terminator).

Reply:
Like this?

```
	char			*safe_desc;
	int			safe_desc_len;

	rman = devinfo_handle_to_rman(res->dr_rman);
	hexmode =  (rman->dm_size > 1000) || (rman->dm_size == 0);
	end = res->dr_start + res->dr_size - 1;

	safe_desc_len = strlen(rman->dm_desc) + 1;
	safe_desc = malloc(safe_desc_len);
	if (safe_desc == NULL) {
		xo_err(1, "No memory!");
		exit(1);
	}
	strlcpy(safe_desc, rman->dm_desc, safe_desc_len);
```

Reply:
Yes like this. Also `safe_desc_len` should be a size_t not an int. I think you can rename the variable just len (safe_desc_len seems too long)

Reply:
Fixed. Thanks!